### PR TITLE
Fix #206.  Use Dates for Program Enrollment Report

### LINKED ISSUE #206: Enlace - Use dates for program enrollment report
Right now, the program enrollment report uses a month/year dropdowns, after which the query doesn't quite make sense.  If you select January of 2018, it only queries information for January 2012, January 2013, and all the Januaries up until 2018.  Replace with dates like the other reports.

### PR DESCRIPTION
**Pull request by frankduncan**
@cecilia-donnelly for your review.

This one can be tested similarly to #204, in the program report.  For a session to be included, both the start and end date need to fall between the date range (and it's optional to leave one of the start or end empty), and then the count is for that session, regardless of when the participant joined.

### CODE REVIEW

File: enlace/reports/num_programs.php [hunk 1]
```diff
@@ -296,20 +272,40 @@ function(idx, display) {
...
+                $results_string = "Results after $start_date_sqlsafe";
+                $date_select_query_str = "Session_Names.Start_Date >= '$start_date_sqlsafe' AND
+                                          Session_Names.End_Date >= '$start_date_sqlsafe'";
+            }
+        }
+        else if ($end_date_sqlsafe != '') {
+            $results_string = "Results before $end_date_sqlsafe";
+            $date_select_query_str = "Session_Names.Start_Date <= '$end_date_sqlsafe'
+                                      Session_Names.End_Date <= '$end_date_sqlsafe'";
+        }
```
Comment (cecilia-donnelly) [inline enlace/reports/num_programs.php]: What if they're both left blank?

File: enlace/reports/num_programs.php [hunk 2]
```diff
@@ -127,7 +103,7 @@ function(idx, display) {
     foreach ($access_array as $program){
         $program_string .= " OR Programs.Program_ID = " . $program;
     }
-            $get_all_programs = "SELECT Session_ID, Session_Name, Name FROM Session_Names INNER JOIN Programs ON Session_Names.Program_ID=Programs.Program_ID " . $program_string . " ORDER BY Name";
+            $get_all_programs = "SELECT Session_ID, Session_Name, Name, Session_Names.End_Date FROM Session_Names INNER JOIN Programs ON Session_Names.Program_ID=Programs.Program_ID " . $program_string . " ORDER BY Name";
```
Comment (cecilia-donnelly) [inline enlace/reports/num_programs.php]: Why not also get start date?

File: enlace/reports/num_programs.php [hunk 3]
```diff
@@ -296,20 +272,40 @@ function(idx, display) {
...
+                                          Session_Names.End_Date >= '$start_date_sqlsafe'";
+            }
+        }
+        else if ($end_date_sqlsafe != '') {
+            $results_string = "Results before $end_date_sqlsafe";
+            $date_select_query_str = "Session_Names.Start_Date <= '$end_date_sqlsafe'
+                                      Session_Names.End_Date <= '$end_date_sqlsafe'";
+        }
+    ?>
+    <h3><?php echo $results_string ?></h3>
```
Comment (cecilia-donnelly) [inline enlace/reports/num_programs.php]: This is nice, thanks!

File: enlace/reports/num_programs.php [hunk 4]
```diff
@@ -296,20 +272,40 @@ function(idx, display) {
...
     <table class="all_projects">
         <tr>
             <th>Program Name</th>
             <th>Session Name</th>
+            <th>Session End Date</th>
```
Comment (cecilia-donnelly) [inline enlace/reports/num_programs.php]: If you're adding end date here (which I think is nice) then I think you should also include start date.

File: enlace/reports/num_programs.php (hunk 1 above)
Comment (cecilia-donnelly) [reply enlace/reports/num_programs.php]: I get zeroes in every column if I leave the dates blank.  I'd expect to get the count of all the participants from the Beginning of Time.

File: enlace/reports/num_programs.php (hunk 4 above)
Comment (frankduncan) [reply enlace/reports/num_programs.php]: Eddie said that they only care about the end date, and didn't want to clutter up the report.

**Comment by frankduncan** [comment]
@cecilia-donnelly I've updated with the changes to fix, I believe!  Thanks!

File: enlace/reports/num_programs.php (hunk 4 above)
Comment (cecilia-donnelly) [reply enlace/reports/num_programs.php]: That's so funny.  Well, okay!

**Pull request merged**